core/chains/orm.go

```diff
@@ -104,6 +123,16 @@ func (o *chainsORM[I, C]) CreateChain(id I, config C, qopts ...pg.QOpt) (chain C
 	return
 }
 
+func (o *chainsORM[I, C]) ensureChain(id I) (err error) {
```

Comment:
This could be a bulk insert maybe which will run faster?

Reply:
Yeah definitely. I'm hestitant to rewrite too much here, but it looks like we might be able to reuse the same query.... :test_tube: